src/nativegasestimator.rs

```diff
@@ -215,8 +215,15 @@ async fn suggest_fee<T: Transport + Send + Sync>(
         return Err(anyhow!("invalid oldest block"));
     };
 
-    let rewards =
-        collect_rewards(transport, oldest_block, fee_history.gas_used_ratio, params).await?;
+    let rewards = match collect_rewards(transport, oldest_block, fee_history.gas_used_ratio, params)
+        .await
+    {
+        Ok(rewards) => rewards,
+        Err(err) => {
+            tracing::warn!(?err, "Failed to collect rewards possibly because node doesn't follow the spec. Proceeding with fallback priority fee.");
```

Comment:
Do you mind splitting the line up, I think `rustfmt` chokes on these very long lines.